packages/design-system/src/components/component-card.tsx

```diff
@@ -0,0 +1,91 @@
+import { forwardRef, type ElementRef, ComponentProps } from "react";
+import { textVariants } from "../";
+import { css, theme } from "../stitches.config";
+
+const cardStyle = css({
+  display: "flex",
+  flexDirection: "column",
+  alignItems: "center",
+  justifyContent: "center",
+  px: theme.spacing[3],
+  width: theme.spacing[19],
+  height: theme.spacing[19],
+  border: `1px solid`,
+  borderColor: theme.colors.borderMain,
+  borderRadius: theme.borderRadius[2],
+  userSelect: "none",
+  color: theme.colors.foregroundIconMain,
+  cursor: "grab",
+  background: theme.colors.backgroundPanel,
+  "&:hover, &[data-state=hover]": {
+    background: theme.colors.backgroundHover,
+  },
+  "&[data-state=disabled]": {
+    background: theme.colors.backgroundPanel,
+    color: theme.colors.foregroundDisabled,
+  },
+  "&:focus-visible, &[data-state=focus]": {
+    outline: `2px solid ${theme.colors.borderFocus}`,
+    outlineOffset: "-2px",
+  },
+  "& svg": {
+    flexGrow: 0,
+    marginTop: theme.spacing[7],
+    width: 22,
+    height: 22,
+  },
+});
+
+const textStyle = css({
+  display: "flex",
+  flexAlign: "center",
+  justifyContent: "center",
+  flexDirection: "column",
+  flexGrow: 0,
+  height: "2em",
+  width: "100%",
+});
+
+const wordStyle = css(textVariants.small, {
+  display: "flex",
+  flexGrow: 0,
+  textAlign: "center",
+  justifyContent: "center",
+  whiteSpace: "nowrap",
+  textOverflow: "ellipsis",
+  overflow: "hidden",
+  width: "100%",
+});
+
+type ComponentCardProps = {
+  label: string;
+  icon: JSX.Element;
+  state?: "hover" | "disabled" | "focus";
+} & ComponentProps<"div">;
+
+export const ComponentCard = forwardRef<ElementRef<"div">, ComponentCardProps>(
+  ({ icon, label, className, state, ...props }, ref) => {
+    return (
+      <div
+        className={cardStyle({ className })}
+        ref={ref}
+        data-state={state}
+        {...props}
+      >
+        {icon}
+        <div className={textStyle()}>
+          {label
+            .split(/\s/)
+            .slice(0, 2)
+            .map((word, index) => (
+              <span className={wordStyle()} key={index}>
+                {word}
+              </span>
+            ))}
```

Comment:
We show only first two words? For example, "Very Important Button" will become "Very Important"?

Reply:
Maybe better line-clamp or text-ellipsis?
https://developer.mozilla.org/en-US/docs/Web/CSS/-webkit-line-clamp

Reply:
basically going with line-clamp is shortening on the total length, where with manual split its shortening each word while making sure we get at least 2 words in if there are 2+, with automatic approach, first word could be long enough so we never see the second word at all

Reply:
<img width="306" alt="Screenshot 2023-03-22 at 15 09 19" src="https://user-images.githubusercontent.com/52824/226930460-5867d842-1619-4af9-a766-02e1013eba1a.png">

vs. 


<img width="392" alt="Screenshot 2023-03-22 at 15 19 57" src="https://user-images.githubusercontent.com/52824/226933393-6643f613-4e5f-4220-a5e6-049c89a7466a.png">

Reply:
The case of 2 long words truncated each is kinda weird and unnatural, not sure if @taylornowotny meant it this way

Reply:
I am trying to get vertically centered and single time truncated work

Reply:
alright got this done better with line clamp, still tradeoffs but seems like better once